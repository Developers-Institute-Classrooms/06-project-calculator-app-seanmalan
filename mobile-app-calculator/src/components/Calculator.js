import { SafeAreaView, View, StyleSheet, Dimensions } from "react-native";
import { React, useState, useEffect } from "react";
import ButtonContainer from "./ButtonContainer";
import OperationDisplay from "./OperationDisplay";
import AsyncStorage from "@react-native-async-storage/async-storage";

const Calculator = () => {
  const [operationDisplay, setOperationDisplay] = useState("");
  const [history, setHistory] = useState([]);
  const [result, setResult] = useState("");
  const [shownResult, setShownResult] = useState("");
  const [firstOperand, setFirstOperand] = useState("");
  const [secondOperand, setSecondOperand] = useState("");
  const [operator, setOperator] = useState("");

  const deviceWidth = Dimensions.get("window").width;
  const deviceHeight = Dimensions.get("window").height;

  useEffect(() => {
    const getData = async () => {
      try {
        const jsonValue = await AsyncStorage.getItem('calculation-history');
        if (jsonValue != null) {
          setHistory(JSON.parse(jsonValue));
        }
      } catch(e) {
        // error reading value
      }
    };

    getData();
  }, []);
  
  

  
useEffect(() => {
const storeData = async (array) => {
  if (array.length > 0) {
    try {
      const jsonValue = JSON.stringify(array);
      await AsyncStorage.setItem('calculation-history', jsonValue);
    } catch (e) {
      
      // saving error
    }
  };
};

    storeData(history);
  }, [history]);



  const isFirstOperand = (value) => {
    return !operator && !operator.includes(value);
  };

  const isSecondOperand = (value) => {
    return operator && !operator.includes(value);
  };

  const operators = ["+", "-", "*", "/"];

  const buttonClicked = (value) => {
    if (value === "DEL") {
      deleteLast();
      return;
    } else if (value === "A/C") {
      onClear();
      return;
    }

    if (operators.includes(value)) {
      if (firstOperand === "") {
        return;
      }
      setOperator(value);
      return;
    }

    if (value === ".") {
      if (firstOperand === "") {
        return;
      } else if (operator !== "" && secondOperand === "") {
        return;
      }


      if (isFirstOperand(value) && firstOperand.includes(".")) {
        return;
      } else if (isSecondOperand(value) && secondOperand.includes(".")) {
        return;
      }
    }


    if (isFirstOperand(value)) {
      setFirstOperand((previousValue) => previousValue + value);
    } else if (isSecondOperand(value)) {
      setSecondOperand((previousValue) => previousValue + value);
    }
  };



  useEffect(() => {
    const settingDisplay = () => {
      if (result !== "") {
        setOperationDisplay(
          `${firstOperand} ${operator} ${secondOperand} = ${result.toFixed(2)}`
        );
        setHistory([...history, `${operationDisplay} = ${result.toFixed(2)}`]);
        setShownResult(
          `${firstOperand} ${operator} ${secondOperand} = ${result.toFixed(2)}`
        );
        clearStates();
      } else {
        setOperationDisplay(`${firstOperand} ${operator} ${secondOperand}`);
      }
    };

    settingDisplay();
  }, [firstOperand, secondOperand, operator, result]);

  const calculate = () => {
    if (firstOperand === "" || secondOperand === "") {
      return;
    }

    if (operator === "+") {
      setResult(Number(firstOperand) + Number(secondOperand));
    } else if (operator === "-") {
      setResult(Number(firstOperand) - Number(secondOperand));
    } else if (operator === "*") {
      setResult(Number(firstOperand) * Number(secondOperand));
    } else if (operator === "/") {
      setResult(Number(firstOperand) / Number(secondOperand));
    }
  };

  const onClear = () => {
    setOperationDisplay("");
    setFirstOperand("");
    setSecondOperand("");
    setOperator("");
    setResult("");
    setShownResult("");
  };

  const deleteLast = () => {
    if (result !== "") {
      onClear();
      return;
    }
    if (secondOperand) {
      setSecondOperand((previousValue) =>
        previousValue.substring(0, previousValue.length - 1)
      );
    } else if (operator) {
      setOperator("");
    } else if (firstOperand) {
      setFirstOperand((previousValue) =>
        previousValue.substring(0, previousValue.length - 1)
      );
    }
  };

  const clearStates = () => {
    setFirstOperand("");
    setSecondOperand("");
    setOperator("");
    setResult("");
  };

  const clearHistory =  async () => {
    setHistory([]);
  };

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <View className="Display" style={{
        flex: 1,
        justifyContent: "flex-end",
        
        marginTop: 20,
        paddingBottom: 30,
        width: deviceWidth,
        height: deviceHeight / 2,
      }}>
        <OperationDisplay
          display={operationDisplay}
          history={history}
          answer={shownResult}
          
        />
      </View>
      <View className="Buttons" testID="button-container" style={{
        flex: 1,
        justifyContent: "center",
        //marginBottom: 10,
        // marginTop: 20,
        width: deviceWidth,
        height: deviceHeight / 2,
        paddingBottom: 30,
      }}>
        <ButtonContainer
          onPress={buttonClicked}
          onClearHistory={clearHistory}
          onClear={onClear}
          onCalculate={calculate}
          onDelete={deleteLast}
          
        />
      </View>
    </SafeAreaView>
  );
};



const styles = StyleSheet.create({
  Display: {
    justifyContent: "flex-end",
    flex: 1,
    marginBottom: 20,
    marginTop: 50,
  },

  Buttons: {
    flex: 1,
    justifyContent: "center",
    marginBottom: 10,
    marginTop: 20,
    height: "80%",
  },
});
export default Calculator;
