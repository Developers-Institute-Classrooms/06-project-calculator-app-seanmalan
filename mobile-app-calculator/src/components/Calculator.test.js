import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import Calculator from "./Calculator";

describe("Calculator", () => {
  test("should display on the screen", () => {
    render(<Calculator />);

    const calculatorHistory = screen.getByText(/Calculation History:/i);
    expect(calculatorHistory).toBeTruthy();

    const currentCalculation = screen.getByText(/Current Calculation:/i);
    expect(currentCalculation).toBeTruthy();

    const buttonContainer = screen.getByTestId("button-container");
    expect(buttonContainer).toBeTruthy();
  });
});

//***************************************** */
test("should update operation display when buttons are clicked", () => {
  const { getByTestId, getByText } = render(<Calculator />);

  const button1 = getByText("1");
  fireEvent.press(button1);
  expect(getByTestId("calculator-display").props.children).toBe("1  ");

  const button2 = getByText("2");
  fireEvent.press(button2);
  expect(getByTestId("calculator-display").props.children).toBe("12  ");

  const buttonPlus = getByText("+");
  fireEvent.press(buttonPlus);
  expect(getByTestId("calculator-display").props.children).toBe("12 + ");

  const button3 = getByText("3");
  fireEvent.press(button3);
  expect(getByTestId("calculator-display").props.children).toBe("12 + 3");

  const buttonEqual = getByText("=");
  fireEvent.press(buttonEqual);
  expect(getByTestId("calculator-display").props.children).toBe(
    "12 + 3 = 15.00"
  );
});

// *****************************************
test("should clear the operation display when the clear button is clicked", () => {
  const { getByTestId, getByText } = render(<Calculator />);

  const button1 = getByText("1");
  fireEvent.press(button1);
  expect(getByTestId("calculator-display").props.children).toBe("1  ");

  const button2 = getByText("2");
  fireEvent.press(button2);
  expect(getByTestId("calculator-display").props.children).toBe("12  ");

  const buttonClear = getByText("A/C");
  fireEvent.press(buttonClear);
  expect(getByTestId("calculator-display").props.children).toBe("");
});

// *****************************************
test("Should delete the last character when the delete button is clicked", () => {
  const { getByTestId, getByText } = render(<Calculator />);

  const button1 = getByText("1");
  fireEvent.press(button1);
  expect(getByTestId("calculator-display").props.children).toBe("1  ");

  const button2 = getByText("2");
  fireEvent.press(button2);
  expect(getByTestId("calculator-display").props.children).toBe("12  ");

  const buttonDelete = getByText("DEL");
  fireEvent.press(buttonDelete);
  expect(getByTestId("calculator-display").props.children).toBe("1  ");
});

// *****************************************
test("Should delete the operand & operator when I press the delete button", () => {
  const { getByTestId, getByText } = render(<Calculator />);

  const button1 = getByText("1");
  fireEvent.press(button1);
  expect(getByTestId("calculator-display").props.children).toBe("1  ");

  const button2 = getByText("2");
  fireEvent.press(button2);
  expect(getByTestId("calculator-display").props.children).toBe("12  ");

  const buttonPlus = getByText("+");
  fireEvent.press(buttonPlus);
  expect(getByTestId("calculator-display").props.children).toBe("12 + ");

  const button7 = getByText("7");
  fireEvent.press(button7);
  expect(getByTestId("calculator-display").props.children).toBe("12 + 7");

  const buttonDelete = getByText("DEL");
  fireEvent.press(buttonDelete);
  expect(getByTestId("calculator-display").props.children).toBe("12 + ");

  const buttonDelete2 = getByText("DEL");
  fireEvent.press(buttonDelete2);
  expect(getByTestId("calculator-display").props.children).toBe("12  ");
});

// *****************************************
test("Should display the answer being pushed into the history array", () => {
  const { getByTestId, getByText } = render(<Calculator />);

  const button1 = getByText("1");
  fireEvent.press(button1);
  expect(getByTestId("calculator-display").props.children).toBe("1  ");

  const button2 = getByText("2");
  fireEvent.press(button2);
  expect(getByTestId("calculator-display").props.children).toBe("12  ");

  const buttonPlus = getByText("+");
  fireEvent.press(buttonPlus);
  expect(getByTestId("calculator-display").props.children).toBe("12 + ");

  const button3 = getByText("3");
  fireEvent.press(button3);
  expect(getByTestId("calculator-display").props.children).toBe("12 + 3");

  const buttonEqual = getByText("=");
  fireEvent.press(buttonEqual);
  expect(getByTestId("calculator-display").props.children).toBe(
    "12 + 3 = 15.00"
  );

  const endHistory = getByText("Calculation History:");
  fireEvent.press(endHistory);
  const history = getByTestId("calculation0");
  expect(history).toBeTruthy();
});



//******************************************* */
//******************************************* */

describe("Operator functions", () => {
  test("Should check the subtract operator functions", () => {
    const { getByTestId, getByText } = render(<Calculator />);

    const button1 = getByText("5");
    fireEvent.press(button1);
    expect(getByTestId("calculator-display").props.children).toBe("5  ");

    const buttonMinus = getByText("-");
    fireEvent.press(buttonMinus);
    expect(getByTestId("calculator-display").props.children).toBe("5 - ");

    const button2 = getByText("2");
    fireEvent.press(button2);
    expect(getByTestId("calculator-display").props.children).toBe("5 - 2");

    const buttonEqual = getByText("=");
    fireEvent.press(buttonEqual);
    expect(getByTestId("calculator-display").props.children).toBe(
      "5 - 2 = 3.00"
    );
  });

  test("Should check the multiply operator functions", () => {

    const { getByTestId, getByText } = render(<Calculator />);

    const button1 = getByText("5");
    fireEvent.press(button1);
    expect(getByTestId("calculator-display").props.children).toBe("5  ");

    const buttonMultiply = getByText("*");
    fireEvent.press(buttonMultiply);
    expect(getByTestId("calculator-display").props.children).toBe("5 * ");

    const button2 = getByText("2");
    fireEvent.press(button2);
    expect(getByTestId("calculator-display").props.children).toBe("5 * 2");

    const buttonEqual = getByText("=");
    fireEvent.press(buttonEqual);
    expect(getByTestId("calculator-display").props.children).toBe(
      "5 * 2 = 10.00"
    );


  })

  test(" Should check the divide operator functions", () => {

    const { getByTestId, getByText } = render(<Calculator />);

    const button1 = getByText("1");
    fireEvent.press(button1);
    expect(getByTestId("calculator-display").props.children).toBe("1  ");

    const button0 = getByText("0");
    fireEvent.press(button0);
    expect(getByTestId("calculator-display").props.children).toBe("10  ");

    const buttonDivide = getByText("/");
    fireEvent.press(buttonDivide);
    expect(getByTestId("calculator-display").props.children).toBe("10 / ");

    const button2 = getByText("2");
    fireEvent.press(button2);
    expect(getByTestId("calculator-display").props.children).toBe("10 / 2");

    const buttonEqual = getByText("=");
    fireEvent.press(buttonEqual);
    expect(getByTestId("calculator-display").props.children).toBe(
      "10 / 2 = 5.00"
    );

  })
});
// it("should clear history when the 'Clear History' button is pressed", async () => {
//   const { getByText, findByTestId } = render(<Calculator />);

//   const clearButton = getByText("Clear History");
//   fireEvent.press(clearButton);

//   // Check if AsyncStorage.clear was called
//     expect(
//       require("@react-native-async-storage/async-storage").clear
//     ).toHaveBeenCalled();
// });
