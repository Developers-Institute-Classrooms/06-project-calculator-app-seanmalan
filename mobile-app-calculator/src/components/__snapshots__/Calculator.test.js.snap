// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View>
    <View
      className="Display"
    >
      <View>
        <Text>
          OperationDisplay
        </Text>
        <Text>
          Current Calculation:
        </Text>
        <Text>
          0
        </Text>
      </View>
    </View>
    <View
      className="Buttons"
    >
      <View>
        <View>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text>
                +
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text>
                -
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text>
                x
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text>
                /
              </Text>
            </View>
          </View>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text>
                0
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text>
                .
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text>
                DEL
              </Text>
            </View>
          </View>
          <View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={1}
            >
              <Text>
                1
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={2}
            >
              <Text>
                2
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={3}
            >
              <Text>
                3
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={4}
            >
              <Text>
                4
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={5}
            >
              <Text>
                5
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={6}
            >
              <Text>
                6
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={7}
            >
              <Text>
                7
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={8}
            >
              <Text>
                8
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "yellow",
                  "borderColor": "black",
                  "borderWidth": 1,
                  "justifyContent": "space-evenly",
                  "width": "100%",
                }
              }
              title={9}
            >
              <Text>
                9
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {},
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                =
              </Text>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "yellow",
                "borderColor": "black",
                "borderWidth": 1,
                "justifyContent": "space-evenly",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text>
              AC
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {},
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Clear History
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Touch Me
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
